From 618ffc7bf963249a8e967bcf424c5d94237c0835 Mon Sep 17 00:00:00 2001
From: Ralph Sennhauser <ralph.sennhauser@gmail.com>
Date: Tue, 21 Oct 2025 09:20:16 +0200
Subject: [PATCH] Require boost-1.69, drop boost_system lib

Boost 1.89 no longer offers the deprecated library for boost_system [1], so
drop it from the list of libraries to link against. This effectively
bumps the minimum required boost version to 1.69 [2].

[1] https://github.com/boostorg/system/commit/7a495bb46d7ccd808e4be2a6589260839b0fd3a3
[1] https://github.com/boostorg/system/commit/2fa0a00583a3a791092568d2ade793314181926e

Fixes: #8471
Signed-off-by: Ralph Sennhauser <ralph.sennhauser@gmail.com>
---
 build/premake/extern_libs5.lua | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/build/premake/extern_libs5.lua b/build/premake/extern_libs5.lua
index f3de8e0ec9..ec2921d115 100644
--- a/build/premake/extern_libs5.lua
+++ b/build/premake/extern_libs5.lua
@@ -234,9 +234,9 @@ extern_lib_defs = {
 			add_default_links({
 				-- The following are not strictly link dependencies on all systems, but
 				-- are included for compatibility with different versions of Boost
-				android_names = { "boost_filesystem-gcc-mt", "boost_system-gcc-mt" },
-				unix_names = { os.findlib("boost_filesystem-mt") and "boost_filesystem-mt" or "boost_filesystem", os.findlib("boost_system-mt") and "boost_system-mt" or "boost_system" },
-				osx_names = { "boost_filesystem", "boost_system" },
+				android_names = { "boost_filesystem-gcc-mt" },
+				unix_names = { os.findlib("boost_filesystem-mt") and "boost_filesystem-mt" or "boost_filesystem" },
+				osx_names = { "boost_filesystem" },
 			})
 		end,
 	},
